src/Annotation/Analyze.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\analyze\Annotation;
+
+use Drupal\Component\Annotation\Plugin;
+
+/**
+ * Defines analyze annotation object.
+ *
+ * @Annotation
+ */
+final class Analyze extends Plugin {
+
+  /**
+   * The plugin ID.
+   */
+  public readonly string $id;
+
+  /**
+   * The human-readable name of the plugin.
+   *
+   * @ingroup plugin_translatable
+   */
+  public readonly string $title;
+
+  /**
+   * The description of the plugin.
+   *
+   * @ingroup plugin_translatable
+   */
+  public readonly string $description;
```

Comment:
Should we make the full report URL / provides full report properties on the annotation?

Reply:
@andrewbelcher I did consider that, but in practice that would mean that a developer could mark the annotation to say that the plugin provided a full report with a custom full report URL but then not implement the code to provide either of them. This way, a plugin is treated as providing a full report and full URL if the methods providing them actually return values. I can see issues with either way so I'm not sure which is best.

Reply:
Yes, that's a good consideration. I'm not 100% convinced of the default (when extending the base) being include the full report URL, but have it empty. But that's definitely something that could be improved in the future rather than now.

Reply:
Yeah it returns a default URL so in the case where someone does want a plugin to show a summary and a full report, all they have to do is implement both render arrays. Otherwise I think we'd have to make them also return a valid URL, which might not be the one we've set a route up for and it would get complicated. Something for future devs to worry about.